common/src/test/java/org/apache/rocketmq/common/UtilAllTests.java

```diff
@@ -0,0 +1,327 @@
+        PowerMockito.when(System.currentTimeMillis())
+                .thenReturn(1515585600000L);
+
+        Assert.assertEquals(1515591000000l,
+                UtilAll.computNextHalfHourTimeMillis());
+    }
+
+    @Test
+    public void testTimeMillisToHumanString2() {
+        Assert.assertEquals("2019-04-25 13:23:34,000",
+                UtilAll.timeMillisToHumanString2(1556195014000l));
+    }
+
+    @Test
+    public void testTimeMillisToHumanString3() {
+        Assert.assertEquals("20190425132334",
+                UtilAll.timeMillisToHumanString3(1556195014000l));
+    }
+
+    @Test
+    public void testGetDiskPartitionSpaceUsedPercent() {
+        Assert.assertEquals(-1,
+                UtilAll.getDiskPartitionSpaceUsedPercent(""), 0);
+        Assert.assertEquals(-1,
+                UtilAll.getDiskPartitionSpaceUsedPercent(null), 0);
+        Assert.assertEquals(-1,
+                UtilAll.getDiskPartitionSpaceUsedPercent("foo"), 0);
+    }
+
+    @Rule
+    private TemporaryFolder folder = new TemporaryFolder();
+
+    @Test
+    public void testGetDiskPartitionSpaceUsedPercent1() throws IOException {
+        File file = folder.newFile("foo.txt");
+
+        Assert.assertNotNull(UtilAll
+                .getDiskPartitionSpaceUsedPercent(file.getPath()));
+    }
+
+    @Test
+    public void testCrc32() {
+        Assert.assertEquals(0, UtilAll.crc32(null));
+        Assert.assertEquals(0, UtilAll.crc32(new byte[]{}));
+        Assert.assertEquals(1438416925, UtilAll.crc32(new byte[]{1, 2, 3}));
+    }
+
+    @Test
+    public void testBytes2string() {
+        Assert.assertEquals("", UtilAll.bytes2string(new byte[]{}));
+        Assert.assertEquals("010203",
+                UtilAll.bytes2string(new byte[]{1, 2, 3}));
+    }
+
+    @Test
+    public void testString2bytes() {
+        Assert.assertNull(UtilAll.string2bytes(""));
+        Assert.assertNull(UtilAll.string2bytes(null));
+
+        Assert.assertArrayEquals(new byte[]{-1}, UtilAll.string2bytes("foo"));
+    }
+
+    @Test
+    public void testUncompress() throws IOException {
+        byte[] by = new byte[]{120, 1, 1, 3, 0, -4, -1, 1, 2, 3, 0, 13, 0, 7};
+        Assert.assertArrayEquals(new byte[]{1, 2, 3}, UtilAll.uncompress(by));
+    }
+
+    @Test
+    public void testCompress() throws IOException {
+        byte[] by = new byte[]{120, 1, 1, 3, 0, -4, -1, 1, 2, 3, 0, 13, 0, 7};
+        Assert.assertArrayEquals(by, UtilAll.compress(new byte[]{1, 2, 3}, 0));
+    }
+
+    @Test
+    public void testAsInt() {
+        Assert.assertEquals(1, UtilAll.asInt("foo", 1));
+        Assert.assertEquals(123, UtilAll.asInt("123", 1));
+    }
+
+    @Test
+    public void testAsLong() {
+        Assert.assertEquals(1l, UtilAll.asLong("foo", 1l));
+        Assert.assertEquals(1556195014000l,
+                UtilAll.asLong("1556195014000", 1l));
+    }
+
+    @Test
+    public void testFormatDate() {
+        Assert.assertEquals("2019-04-25 13:23:34", UtilAll
```

Comment:
Maybe break the line after `UtilAll.formatDate(`?

Reply:
Done